app/models/audit/event/authn/role_id.rb

```diff
@@ -0,0 +1,23 @@
+module Audit
+  module Event
+    class Authn
+      # RoleId, in the context of authentication, is subtly different from an
+      # ordinary "role_id".  This is because an authn event can fail when the
+      # user or host authenticating doesn't exist.  In this case, the "role_id"
+      # becomes "what the role id would be if the user _did_ exist". In all
+      # other cases, it's the actual "role_id" for the existing user.
+      #
+      # The control parameter smell here is intentional: This is the object
+      # that is hiding the "nil"-ugliness.
+      # :reek:ControlParameter
+      class RoleId
+        # Note: "username" may refer to a user or host.
+        def self.new(role:, account:, username:)
```

Comment:
I don't particularly like this method being called `new` because naively I would expect it to act like a constructor and return an object of type `Audit::Event::Authn::RoleId`. I'm not sure I have a better idea, though.

Reply:
It's a little funny, true.  There are two alternatives here:

1. Make it a utility function.  I am not a big fan of utility functions in ruby so lean away from this.
2. Have it return a proper `RoleId` instance, which has a `to_s` method which returns the the actual role id -- ie, what we're returning in the above.  Conceptually, this is cleaner and more in the OO / ruby spirit.  Wdyt?

Reply:
Yeah I think 2 makes sense. It's a little more clear what's going on in that case.